plone/releaser/pip.py

```diff
@@ -0,0 +1,178 @@
+        # uses them.
+        newline = f"{package_name}=={new_version}"
+        if contents.count(" == ") > contents.count("==") / 2:
+            newline = newline.replace("==", " == ")
+        if package_name not in self:
+            contents += newline + "\n"
+
+        reg = re.compile(
+            rf"^{package_name} ?==.*$",
+            re.MULTILINE,
+        )
+        new_contents = reg.sub(newline, contents)
+        self.path.write_text(new_contents)
+
+    def get(self, package_name):
+        return self.__getitem__(package_name)
+
+    def set(self, package_name, new_version):
+        return self.__setitem__(package_name, new_version)
+
+
+class IniFile(UserDict):
+    """Ini file for mxdev.
+
+    What we want to do here is similar to what we have in buildout.py
+    in the CheckoutsFile: remove a package from auto-checkouts.
+    For mxdev: set 'use = false'.
+    The default is in 'settings': 'default-use'.
+    """
+    def __init__(self, file_location):
+        self.file_location = file_location
+        self.path = pathlib.Path(self.file_location).resolve()
+        self.config = ConfigParser(
+            default_section="settings",
+            interpolation=ExtendedInterpolation(),
+        )
+        with open(self.file_location) as f:
+            self.config.read_file(f)
+        self.default_use = to_bool(self.config["settings"].get("default-use", True))
+
+    @property
+    def data(self):
+        checkouts = {}
+        for package in self.config.sections():
+            use = to_bool(self.config[package].get("use", self.default_use))
+            if use:
+                # Map from lower case to actual case, so we can find the package.
+                checkouts[package.lower()] = package
+        return checkouts
+
+    def __contains__(self, package_name):
+        return package_name.lower() in self.data
+
+    def __setitem__(self, package_name, enabled=True):
+        """Enable or disable a checkout.
+
+        Mostly this will be called to disable a checkout.
+        Expected is that default-use is false.
+        This means we can remove 'use = true' from the package.
+
+        But let's support the other way around as well:
+        when default-use is true, we set 'use = false'.
+        """
+        stored_package_name = self.data.get(package_name.lower())
+        if stored_package_name:
+            package_name = stored_package_name
+            use = to_bool(self.config[package_name].get("use", self.default_use))
+        else:
+            use = False
+        if use and enabled:
+            print(f"{package_name} is already used as checkout.")
+            return
+        if not use and not enabled:
+            print(f"{package_name} is not used as checkout.")
+            return
+
+        contents = self.path.read_text()
+        if not contents.endswith("\n"):
+            contents += "\n"
+
+        lines = []
+        found_package = False
+        for line in contents.splitlines():
```

Comment:
It might be possible to tell configparser or mxdev to change the parsed config.  But then you have to write it out.  And this likely means that unintended changes occur, like removing comments.

Reply:
Oh rigut, comments... 👍🏾